### PR TITLE
Bump Promises dependency

### LINKED ISSUE #8334: Upgrade Promise dependency before Xcode 13 is released
### [REQUIRED] Step 1: Describe your environment

  * Xcode version: 13 beta 2
  * Firebase SDK version: 8.2.0
  * Installation method: `CocoaPods`
  * Firebase Component: Core

### [REQUIRED] Step 2: Describe the problem

The current version of Xcode 13 (beta 2) have a change to the swift compiler which makes `await` into a keyword. It seems this change is not guarded be a feature flag, so anything compiled with Xcode 13 (even if not targeting new versions) will not be able to use `await` as a function name (See [0296-async](https://github.com/apple/swift-evolution/blob/main/proposals/0296-async-await.md#source-compatibility) ).

This conflicts with [google/promise](https://github.com/google/promises) which Firebase depends on (through `GoogleUtilities/Environment`).

`Google/Promises` have been updated with a fix that renames `await`, but required a major release (v2.0.0).

Since Firebase is still depending on the old version, any project that use both Firebase and Promise will not be able to compile with Xcode 13 since there's no way to allow two versions of Promise in one project.

#### Steps to reproduce:

1. Create a project that Use both Firebase and Promise and compile it with Xcode 13
2. Trying to use the old Promise will result in compile errors
3. Updating to use Promise 2 is not possible because of firebase dependencies:
```
        FirebaseCore (~> 8.0) was resolved to 8.2.0, which depends on
          GoogleUtilities/Environment (~> 7.4) was resolved to 7.4.3, which depends on
            PromisesObjC (~> 1.2)
```

### PR DESCRIPTION
**Pull request by ncooke3**
Plan is to:
- merge this PR
- wait for clean nightlies
- then publish GDT & GUL minor releases

Fixes #8334

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -153,7 +153,7 @@ let package = Package(
     .package(
       name: "GoogleUtilities",
       url: "https://github.com/google/GoogleUtilities.git",
-      "7.4.0" ..< "8.0.0"
+      "7.5.0" ..< "8.0.0"
     ),
```
Comment (ncooke3) [inline Package.swift:157]: GoogleUtilities should bump to its new minor right?

**Comment by paulb777** [comment]
Test the podspec changes first, to avoid the need to prematurely tag the GoogleUtilities and GTM repos.

The podspecs should be updated.

Bumping the GoogleUtilities minimum dependency is fine, but not required since the firebase-ios-sdk SPM should still work with 7.4.x ....

**Comment by ncooke3** [comment]
@paulb777 So with the specs bumped and CI green,  GDT and GULs are good to publish, correct?

**Comment by paulb777** [comment]
Let's wait for a clean nightly run before publishing GDT and GoogleUtilities.

This is good to merge after adding a CHANGELOG update for 8.4.0

**Review by paulb777 — APPROVED** [review]

**Pull request merged**